--- jquery_update.install	2009-04-24 11:54:57.000000000 +0200
+++ jquery_update.install	2009-04-24 11:55:22.000000000 +0200
@@ -48,9 +48,13 @@ function jquery_update_update_6100() {
  * version update to minified.
  */
 function jquery_update_update_6200() {
+  $ret = array();
+
   if (variable_get('jquery_update_compression_type', 'min') == 'pack') {
     variable_set('jquery_update_compression_type', 'min');
   }
+
+  return $ret;
 }

 /**
